sdk/python/kfp/components/component_factory.py

```diff
@@ -321,6 +319,10 @@ def extract_component_interface(
                 signature.return_annotation)
             output_spec = structures.OutputSpec(type=type_struct)
             outputs[output_name] = output_spec
+    elif return_ann != inspect.Parameter.empty and return_ann != structures.ContainerSpec:
+        raise TypeError(
+            'Return annotation should be either ContainerSpec or ignored for container components.'
```

Comment:
nit: Consider `omitted` instead of `ignored`. Ignored sounds like the KFP SDK ignores it, but we're actually talking about the author here.